eras/conway/impl/cddl/lib/Cardano/Ledger/Conway/HuddleSpec.hs

```diff
@@ -0,0 +1,1256 @@
+drepVotingThresholdsRule p =
+  "drep_voting_thresholds"
+    =:= arr
+      [ a (huddleRule @"unit_interval" p) //- "motion no confidence"
+      , a (huddleRule @"unit_interval" p) //- "committee normal"
+      , a (huddleRule @"unit_interval" p) //- "committee no confidence"
+      , a (huddleRule @"unit_interval" p) //- "update constitution"
+      , a (huddleRule @"unit_interval" p) //- "hard fork initiation"
+      , a (huddleRule @"unit_interval" p) //- "PP network group"
+      , a (huddleRule @"unit_interval" p) //- "PP economic group"
+      , a (huddleRule @"unit_interval" p) //- "PP technical group"
+      , a (huddleRule @"unit_interval" p) //- "PP governance group"
+      , a (huddleRule @"unit_interval" p) //- "treasury withdrawal"
+      ]
+
+conwayMultiasset ::
+  forall era a.
+  (HuddleRule "policy_id" era, HuddleRule "asset_name" era, IsType0 a) =>
+  Proxy era ->
+  a ->
+  GRuleCall
+conwayMultiasset p =
+  binding $ \x ->
+    "multiasset"
+      =:= mp
+        [ 0
+            <+ asKey (huddleRule @"policy_id" p)
+            ==> mp [1 <+ asKey (huddleRule @"asset_name" p) ==> x]
+        ]
+
+conwayValueRule ::
+  forall era.
+  ( HuddleRule "policy_id" era
+  , HuddleRule "asset_name" era
+  , HuddleRule "positive_coin" era
+  ) =>
+  Proxy era ->
+  Rule
+conwayValueRule p =
+  "value"
+    =:= huddleRule @"coin" p
+    / sarr [a $ huddleRule @"coin" p, a $ conwayMultiasset p (huddleRule @"positive_coin" p)]
+
+conwayMintRule ::
+  forall era.
+  ( HuddleRule "policy_id" era
+  , HuddleRule "asset_name" era
+  , HuddleRule "nonzero_int64" era
+  ) =>
+  Proxy era ->
+  Rule
+conwayMintRule p =
+  "mint"
+    =:= mp
+      [ 1
+          <+ asKey (huddleRule @"policy_id" p)
+          ==> mp [1 <+ asKey (huddleRule @"asset_name" p) ==> huddleRule @"nonzero_int64" p]
+      ]
+
+conwayWithdrawalsRule :: forall era. Era era => Proxy era -> Rule
+conwayWithdrawalsRule p =
+  "withdrawals"
+    =:= mp
+      [ 1
+          <+ asKey (huddleRule @"reward_account" p)
+          ==> huddleRule @"coin" p
+      ]
+
+conwayRedeemerTag :: Rule
+conwayRedeemerTag =
+  comment
+    [str|0: spend
+        |1: mint
+        |2: cert
+        |3: reward
+        |4: voting
+        |5: proposing
+        |]
+    $ "redeemer_tag"
+      =:= (0 :: Integer)
+      ... (5 :: Integer)
+
+conwayRedeemer ::
+  forall era.
+  ( HuddleRule "redeemer_tag" era
+  , HuddleRule "plutus_data" era
+  , HuddleRule "ex_units" era
+  ) =>
+  Proxy era ->
+  Rule
+conwayRedeemer p =
+  "redeemer"
+    =:= arr
+      [ "tag" ==> huddleRule @"redeemer_tag" p
+      , "index" ==> (VUInt `sized` (4 :: Word64))
+      , "data" ==> huddleRule @"plutus_data" p
+      , "ex_units" ==> huddleRule @"ex_units" p
+      ]
+
+instance HuddleRule "min_int64" ConwayEra where
+  huddleRule _ = minInt64Rule
+
+instance HuddleRule "max_int64" ConwayEra where
+  huddleRule _ = maxInt64Rule
+
+instance HuddleRule "int64" ConwayEra where
+  huddleRule = int64Rule @ConwayEra
```

Comment:
These too can be removed from individual eras and added either to `cardano-ledger-core` or better to the first era they were used in and made available for all eras.
`max_int64` can never change its values. In the matter of fact we should prevent it from changing, since this things like `max_int64` universally have exact same value.